src/test/java/com/google/ehub/servlets/ProfileServletTest.java

```diff
@@ -0,0 +1,194 @@
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.ehub.data.ProfileDatastore;
+import com.google.ehub.data.UserProfile;
+import com.google.ehub.servlets.LoginServlet;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class ProfileServletTest {
+  private static final String NAME_PROPERTY_KEY = "name";
+  private static final String EMAIL_PROPERTY_KEY = "email";
+  private static final String USERNAME_PROPERTY_KEY = "username";
+  private static final String BIO_PROPERTY_KEY = "bio";
+  private static final String EDIT_PROPERTY_KEY = "edit";
+  private static final String FALSE_EDIT_VALUE = "false";
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String PROFILE_ITEM_KIND = "profile";
+  private static final String NAME = "Honey";
+  private static final String EMAIL = "honey7100@gmail.com";
+  private static final String USERNAME = "honey7100";
+  private static final String BIO = "My name means honey!";
+  private static final String REDIRECT = "/ProfilePage.html";
+  private static final String REDIRECT_GET = "/CreateProfilePage.html";
+  private static final String ERROR_MESSAGE = "User must logged in";
+  private static final String POST_ERROR_MESSAGE = "Post Request parameters empty";
+  private final ProfileServlet servlet = new ProfileServlet();
+  private final ProfileDatastore profileData = new ProfileDatastore();
+
+  private LocalServiceTestHelper helper = new LocalServiceTestHelper(
+      new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvEmail(EMAIL)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail.com");
+  private final DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+  @Mock PrintWriter printWriter;
+  @Mock UserService userService;
```

Comment:
Does the ```userService``` really need to be mocked?

Reply:
I actually don't need it since it's not being used